src/contracts/fund/participation/Participation.ts

```diff
@@ -181,17 +219,70 @@ export class Participation extends Contract {
    */
   public executeRequestFor(from: Address, forWhom: Address) {
     const amgu = this.calculateAmgu.bind(this);
-    const incentive = () => Promise.resolve(new BigNumber(10).exponentiatedBy(16));
 
-    const validate = async () => {};
+    const validate = async () => {
+      if (!(await this.hasRequest(from))) {
+        throw new NoInvestmentRequestError(from);
+      }
+
+      const request = await this.getRequest(from);
+      const priceSource = new PriceSourceInterface(this.environment, await this.getPriceSource());
+      if (!priceSource.hasValidPrice(request.investmentAsset)) {
```

Comment:
This is checking the promise, not the result of the promise. Has to be wrapped with `await`.